lib/vagrant-openshift/hooks/configure_aws.rb

```diff
@@ -0,0 +1,54 @@
+#--
+# Copyright 2016 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#++
+
+require_relative "../aws"
+
+module Vagrant
+  module Openshift
+    module Hooks
+      class ConfigureAWS
+        def initialize(app, env)
+          @app = app
+        end
+
+        def call(env)
+          if env[:machine].config.openshift.autoconfigure_aws == true
+            aws_creds = Vagrant::Openshift::Aws::aws_creds()
+            raise "AWS credentials not configured" if aws_creds.nil?
+
+            aws = env[:machine].config.vm.get_provider_config(:aws)
+            if !aws.nil?
+              aws.access_key_id = aws_creds["AWSAccessKeyId"]
+              aws.secret_access_key = aws_creds["AWSSecretKey"]
+              aws.keypair_name = aws_creds["AWSKeyPairName"]
+
+              box_info = YAML.load_file(Pathname.new(File.expand_path("#{__FILE__}/../../templates/command/init-openshift/box_info.yaml")))
+              if !aws.ami.start_with?("ami-")
+                os, stage = aws.ami.split(":", 2)
+                env[:machine].config.ssh.username = box_info[os.to_sym][stage.to_sym][:aws][:ssh_user]
+              end
+            end
+            env[:machine].config.ssh.private_key_path = [aws_creds["AWSPrivateKeyPath"]]
+            env[:machine].config.vm.box = "https://github.com/mitchellh/vagrant-aws/raw/master/dummy.box"
+            env[:machine].config.vm.synced_folders["/vagrant"] = {:disabled => true, :guestpath => "/vagrant", :hostpath => "."}
```

Comment:
Yes; the reason I left them in is because of the recommendation for disabling the default synced folder at https://www.vagrantup.com/docs/synced-folders/basic_usage.html#disabling :

Adding `config.vm.synced_folder ".", "/vagrant", disabled: true` to a Vagrantfile as recommended at the above location results in the hash `{:disabled => true, :guestpath => "/vagrant", :hostpath => "."}` being stored, rather just than `{:disabled => true}`.  I figured, this being Vagrant, that not shortcutting may be less likely to cause an issue somewhere else.